src/combinators/mod.rs

```diff
@@ -1017,6 +1017,24 @@ impl BackToTheBuffer for &mut [u8] {
     }
 }
 
+impl BackToTheBuffer for Vec<u8> {
+    fn reserve_write_use<Tmp, Gen: Fn(Self) -> GenResult<(Self, Tmp)>, Before: Fn(Self, Tmp) -> GenResult<Self>>(mut self, reserved: usize, gen: Gen, before: Before) -> GenResult<Self> {
+        let start_len = self.len();
+        self.reserve(reserved);
+        unsafe { self.set_len(start_len + reserved) };
+        let tmp_vec = unsafe { Vec::from_raw_parts(self[start_len..].as_mut_ptr(), 0, reserved) };
+        let (vec, tmp) = gen(self)?;
+        let tmp_vec = before(tmp_vec, tmp)?;
+        let tmp_written = tmp_vec.len();
+        // forget our hacky tmp vec not to free this sub-vec twice (once for tmp_vec, once for vec)
+        core::mem::forget(tmp_vec);
+        if tmp_written != reserved {
```

Comment:
Here we definitely need to panic. Because of the problem above if writing more, and also if less was written then there's uninitialized memory now.

Reply:
Yes, I pushed the panic right after the initial commit